### ISSUE TITLE
Inference by HuggingFace 代码问题

### ISSUE THREAD

**Issue by SongWWWWWW** [zjunlp/OceanGPT#2]
我遇到了"The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results."的问题, 
修改代码为：

from transformers import AutoModelForCausalLM, AutoTokenizer
import torch

device = "cuda:0" # the device to load the model onto
path = 'zjunlp/OceanGPT-14B-v0.1'
device_map="cuda:0"
model = AutoModelForCausalLM.from_pretrained(
    path,
    torch_dtype=torch.bfloat16,
    device_map="auto"
)
tokenizer = AutoTokenizer.from_pretrained(path)

prompt = "Which is the largest ocean in the world?"
messages = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": prompt}
]
text = tokenizer.apply_chat_template(
    messages,
    tokenize=False,
    add_generation_prompt=True
)
model_inputs = tokenizer([text], return_tensors="pt").to(device)
attention_mask = model_inputs['attention_mask'] # 加入代码
generated_ids = model.generate(
    model_inputs.input_ids,
    attention_mask = attention_mask, # 加入代码
    max_new_tokens=512
)
generated_ids = [
    output_ids[len(input_ids):] for input_ids, output_ids in zip(model_inputs.input_ids, generated_ids)
]

response = tokenizer.batch_decode(generated_ids, skip_special_tokens=True)[0]
print(response)

**Comment by coderxyd** [comment]
When I use the latest transformers, the code above can run normally. You can check your transformers version and try upgrading to the latest transformers.

**Comment by SongWWWWWW** [comment]
yes，the Transformers library version is somewhat behind the latest release.
